knimin/lib/data_access.py

```diff
@@ -2415,6 +2415,126 @@ def set_deposited_ebi(self):
                  WHERE barcode IN %s"""
         self._con.execute(sql, [barcodes])
 
+    def create_study(self, study_id, title=None, alias=None, notes=None):
+        """Creates a study
+
+        Parameters
+        ----------
+        study_id : int
+            Positive if it is a Qiita study ID, negative if others
+        title : str
+        alias : str
+        notes : str
+
+        Raises
+        ------
+        ValueError
+            If there is error creating the study
+        """
+        with self._con.cursor() as cur:
+            cur.execute('BEGIN')
```

Comment:
I don't think this is working as expected. `execute_fetchone` explicitly executes a commit, which means that those calls are either (1) executed in a different "transaction" since it is using a different cursor or (2) everything is committed as before.

I think the best path moving forward is actually copying the transaction object in Qiita to here (probably under a different module to avoid conflicts) and use it here, rather than implementing different transactions here, which is just gonna make the code less compliant with the other projects in the lab.